packages/headless/src/api/search/search-api-client.ts

```diff
@@ -254,6 +268,12 @@ function isSuccessPlanResponse(
   );
 }
 
+function isSuccessHtmlResponse(
+  r: PlatformResponse<string>
+): r is PlatformResponse<string> {
+  return typeof r.body === 'string';
```

Comment:
Why do we need that "special check" for Quickview ? It seems very flaky as a check.

The error code does not work ?

Reply:
I was following the pattern in place. For all endpoints, we check whether the payload is successful, rather than if the status code matches an error code. I did not dig into why this is, and honestly, I think it is more robust to use the status code.

_Unless_ the intent is that a successful response that is empty should be treated as an error? I do not see an immediate reason for why this would be preferred though.

I could revisit this in a separate jira and refactor if it makes sense, but for this PR, I prefer to keep the same pattern ...